core/src/main/java/org/apache/accumulo/core/file/rfile/PrintInfo.java

```diff
@@ -192,23 +268,43 @@ public void execute(final String[] args) throws Exception {
           }
         }
 
+        Formatter formatter = null;
+        if (opts.formatterClazz != null) {
+          final Class<? extends Formatter> formatterClass = getFormater(opts.formatterClazz);
+          FormatterConfig config = new FormatterConfig();
+          config.setPrintTimestamps(true);
+          formatter =
+              FormatterFactory.getFormatter(formatterClass, new FormatterAdapter(dataIter), config);
+        }
+
         for (String lgName : localityGroupCF.keySet()) {
           LocalityGroupUtil.seek(dataIter, new Range(), lgName, localityGroupCF);
-          while (dataIter.hasTop()) {
-            Key key = dataIter.getTopKey();
-            Value value = dataIter.getTopValue();
-            if (opts.dump) {
-              System.out.println(key + " -> " + value);
-              if (System.out.checkError())
+
+          if (formatter != null) {
+            while (formatter.hasNext()) {
```

Comment:
This seems a bit dicey.  The underlying iterator is reset under the formatter for each locality group.  If formatter were stateful with hasNext() this may cause problems.

Reply:
Yeah, I wasn't too happy about this one either. I suspect one approach would be to create a new instance of the formatter for each locality group.